src/sentry/tasks/backfill_seer_grouping_records.py

```diff
@@ -0,0 +1,312 @@
+    redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
+    last_processed_id = int(redis_client.get(LAST_PROCESSED_REDIS_KEY) or 0)
+    for group_id_batch in chunked(
+        RangeQuerySetWrapper(
+            Group.objects.filter(
+                project_id=project_id, id__gt=last_processed_id, type=1
+            ).values_list("id", flat=True),
+            result_value_getter=lambda item: item,
+        ),
+        ITERATOR_CHUNK,
+    ):
+        query = Query(
+            match=events_entity,
+            select=[
+                Column("group_id"),
+                Function("max", [Column("event_id")], "event_id"),
+                Column("message"),
+            ],
+            groupby=[Column("group_id"), Column("message")],
+            where=[
+                Condition(Column("project_id"), Op.EQ, project_id),
+                Condition(Column("group_id"), Op.IN, group_id_batch),
+                Condition(
+                    Column("timestamp", entity=events_entity), Op.GTE, time - timedelta(days=90)
+                ),
+                Condition(Column("timestamp", entity=events_entity), Op.LT, time),
+            ],
+            orderby=[OrderBy(Column("group_id"), Direction.ASC)],
+        )
+
+        request = Request(
+            dataset=Dataset.Events.value,
+            app_id=Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value,
+            query=query,
+            tenant_ids={
+                "referrer": Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value,
+                "organization_id": organization_id,
+            },
+        )
+        result = bulk_snuba_queries(
+            [request], referrer=Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value
+        )
+
+        if result and result[0].get("data"):
+            rows: list[GroupEventRow] = result[0]["data"]
+            data = lookup_group_data_stacktrace_bulk_with_fallback(project, rows)
+            response = post_bulk_grouping_records(
+                CreateGroupingRecordsRequest(
+                    group_id_list=group_id_batch,
+                    data=data["data"],
+                    stacktrace_list=data["stacktrace_list"],
+                )
+            )
+            if response["success"]:
+                redis_client.set(
+                    f"{LAST_PROCESSED_REDIS_KEY}", group_id_batch[-1], ex=60 * 60 * 24 * 7
+                )
+                groups = Group.objects.filter(id__in=group_id_batch)
+                for group in groups:
+                    if group.data.get("metadata"):
+                        group.data["metadata"].update({"has_embeddings_record_v1": True})
+                    else:
+                        group.data["metadata"] = {"has_embeddings_record_v1": True}
+                num_groups_records_created += Group.objects.bulk_update(groups, ["data"])
+
+    return num_groups_records_created
+
+
+def lookup_group_data_stacktrace_bulk_with_fallback(
```

Comment:
this function and the following functions were based off of this [backfill script](https://github.com/getsentry/getsentry/blob/b2689f4e98cb5d641ecdda4605e9cf84932b2a67/bin/backfill_perf_issue_events_issue_platform.py#L476)